SPEC.md

```diff
@@ -314,6 +324,36 @@ String is either a [Quoted String](#quoted-string) or a
 [Raw String](#raw-string). Quoted Strings may include escaped characters, while
 Raw Strings always contain only the literal characters that are present.
 
+Strings _MUST_ be represented as UTF-8 values.
+
+Strings _MUST NOT_ include the code points for [disallowed literal
+code points](#disallowed-literal-code-points) directly. If needed, they can be
+specified with their corresponding `\u{}` escape.
+
+### Multi-line Strings
+
+Strings may span multiple lines with literal Newlines, in which case the
+resulting String is "dedented" according to the line with the fewest number of
```

Comment:
Not "fewest number", that would count a two spaces on one line the same as two tabs on the next. Should instead find the largest exactly shared whitespace prefix.

Reply:
this is actually the intention: for two tabs to be treated the same as two spaces, as far as indentation goes. I'm not sure we should be making any decisions about how many spaces wide a tab is?

Reply:
Actually that brings up a good point, what happens if someone mixes spaces and tabs into a multiline string? 

How do you even dedent that?

Reply:
like the text says: based on the number of characters. `\t\t` and `\s\s` count as two, and both mean that you remove "two items of whitespace" from the other lines, regardless of whether they're tabs or spaces.

Reply:
The JS proposal did a survey of langs; several *do* just count the characters (and one treats tabs as equivalent to a number of spaces), and the champions explicitly didn't like those and went with exact prefix match. I also prefer that; it's weird that the following will dedent both lines to have zero whitespace:

```
node "
    spaced
				tabbed
"
```

> I'm not sure we should be making any decisions about how many spaces wide a tab is?

Correct, we shouldn't; exact ws prefix match only. So my example above will *not dedent at all*, since the common ws prefix is nothing.

People shouldn't be mixing tabs and spaces between lines, but if they do, I *guarantee* they don't expect one tab to equal one space for removal. Dedenting is meant to achieve a *visual* goal.

Reply:
@tabatkins what do you think of doing what @IceDragon200 suggested [here](https://github.com/kdl-org/kdl/pull/286#issuecomment-1854704718), and, instead of looking for "common prefix", doing "strip whitespace matching the whitespace between the final newline and the closing `"`"?

That would allow literal multiline strings to be themselves fully-indented, and give a clear, single point that users could adjust to set the desired indentation of their strings, without weird `\s` trickery?

Reply:
Yeah I'm fine with that. I lean *slightly* towards matching JS `dedent` just because it makes reasonable decisions overall here and matching precedent is nice, but if you prefer "last line sets the indent, and it's a parse error for any other non-empty line to not have an exactly matching prefix", I'm okay with that.

Reply:
The only thing that convinces me to do last-line instead of common-prefix is the ability to easily control overall-final-indent, which seems valuable. So I think I'll go with that...

Reply:
I've gone ahead and edited the PR (and associated tests) to match this; the final line now gives the whitespace to remove. We'll want a bunch more tests for this, but #361 is already covering that.